NeoML/Python/neoml/Dnn/Split.py

```diff
@@ -20,7 +20,48 @@
 import numpy
 
 
-class SplitChannels(Layer):
+class SplitLayer(Layer):
+    """The base (abstract) class for a split layer.
+    """
+    def __init__(self, classname, input_layer, sizes, name):
+        assert hasattr(PythonWrapper, classname), 'Incorrect split layer specified: ' + classname
+
+        if type(input_layer) is getattr(PythonWrapper, classname):
+            super().__init__(input_layer)
+            return
+
+        layers, outputs = check_input_layers(input_layer, 1)
+
+        internal = getattr(PythonWrapper, classname)(str(name), layers[0], int(outputs[0]), self.__sizes_to_array(sizes))
+        super().__init__(internal)
+
+    @property
+    def output_sizes(self):
+        """
+        """
+        return self._internal.get_output_counts()
+
+    @output_sizes.setter
+    def output_sizes(self, value):
+        """
+        """        
+        self._internal.set_output_counts(self.__sizes_to_array(value))
+
+    @staticmethod
+    def __sizes_to_array(sizes) -> numpy.ndarray:
+        sizes = numpy.array(sizes, dtype=numpy.int32)
+        if sizes.ndim != 1 or sizes.size > 3:
```

Comment:
Why do you have the upper limit of `3` here? It should be up to `31`...

Reply:
I see, it was in the original layer. But that's a mistake! It should be 31. (MathEngine can handle split/merge with up to 32 parts).